src/realm/index_string.cpp

```diff
@@ -323,7 +323,8 @@ namespace {
 
 // replicates the 4 least significant bits each times 8
 // eg: abcd -> aaaaaaaabbbbbbbbccccccccdddddddd
-int32_t replicate_4_lsb_x8(int32_t i) {
+uint32_t replicate_4_lsb_x8(uint32_t i)
```

Comment:
This function overflows for a lot of valid inputs when done on a signed integer. Changing this to unsigned doesn't actually modify the emitted code.